File: hard/3_three_number_sum/three_number_sum.test.js
const  threeNumberSum = require('./three_number_sum');

describe("Three Number Sum", () => {
	it("Test Case #1", () => {
		const queries = [12, 3, 1, 2, -6, 5, -8, 6];
		const target = 0;
		const expected = [[-8, 2, 6], [-8, 3, 5], [-6, 1, 5]];

		const actual = threeNumberSum(queries, target);
		expect(actual).toEqual(expected);
	});

	it("Test Case #2", () => {
		const queries = [1, 2, 3];
		const target = 6;
		const expected = [[1, 2, 3]];

		const actual = threeNumberSum(queries, target);
		expect(actual).toEqual(expected);
	});
});

File: hard/3_three_number_sum/three_number_sum_test.go
package main

import (
	"github.com/stretchr/testify/require"
	"testing"
)

func TestThreeNumberSum(t *testing.T) {
	queries := []int{12, 3, 1, 2, -6, 5, -8, 6}
	target := 0
	expected := [][]int{{-8, 2, 6}, {-8, 3, 5}, {-6, 1, 5}}

	actual := ThreeNumberSum(queries, target)
	require.Equal(t, expected, actual)

	queries = []int{1, 2, 3}
	target = 6
	expected = [][]int{{1, 2, 3}}
	actual = ThreeNumberSum(queries, target)
	require.Equal(t, expected, actual)
}
